messaging/relayer/internal/msgobserver/eth/event_watcher.go

```diff
@@ -57,22 +60,122 @@ func (l *SFCCrossCallWatcher) Watch() {
 	l.start(sub, chanEvents)
 }
 
-func (l *SFCCrossCallWatcher) start(sub event.Subscription, chanEvents <-chan *functioncall.SfcCrossCall) {
+func (l *SFCCrossCallRealtimeEventWatcher) start(sub event.Subscription, chanEvents <-chan *functioncall.SfcCrossCall) {
 	logging.Info("Start watching %v...", l.SfcContract)
 	for {
 		select {
 		case err := <-sub.Err():
 			// TODO: communicate this to the calling context
 			logging.Error("error in log subscription %v", err)
-		case log := <-chanEvents:
-			l.Handler.Handle(log)
+		case ev := <-chanEvents:
+			l.Handler.Handle(ev)
 		case <-l.end:
 			logging.Info("Stop watching %v.", l.SfcContract)
 			return
 		}
 	}
 }
 
-func NewSFCCrossCallWatcher(context context.Context, handler EventHandler, contract *functioncall.Sfc, end chan bool) *SFCCrossCallWatcher {
-	return &SFCCrossCallWatcher{EventWatcherConfig: EventWatcherConfig{Context: context, Handler: handler}, SfcContract: contract, end: end}
+func NewSFCCrossCallRealtimeEventWatcher(context context.Context, handler EventHandler, contract *functioncall.Sfc, end chan bool) *SFCCrossCallRealtimeEventWatcher {
+	return &SFCCrossCallRealtimeEventWatcher{EventWatcherConfig: EventWatcherConfig{Context: context, Handler: handler}, SfcContract: contract, end: end}
+}
+
+type BlockHeadProducer interface {
+	SubscribeNewHead(ctx context.Context, ch chan<- *types.Header) (ethereum.Subscription, error)
+}
+
+// SFCCrossCallFinalisedEventWatcher listens for events from a Simple Function Call bridge and returns processes these events only once they are
```

Comment:
I think the wording is wrong here. Do you mean "returns" or "processes"?

Reply:
thanks